src/audio/nimio-processor.js

```diff
@@ -44,13 +48,37 @@ class AudioNimioProcessor extends AudioWorkletProcessor {
     this._latencyCtrl.speedFn = this._setSpeed.bind(this);
 
     if (!this._idle) {
-      this._audioBuffer = new ReadableAudioBuffer(
-        options.processorOptions.audioSab,
-        options.processorOptions.capacity,
-        this._sampleRate,
-        this._channelCount,
-        this._fSampleCount,
-      );
+      this._useSharedAudioSab =
+        options.processorOptions.audioSabShared !== false;
+      this._portFramesReceived = 0;
+      if (this._useSharedAudioSab && options.processorOptions.audioSab) {
+        this._audioBuffer = new ReadableAudioBuffer(
+          options.processorOptions.audioSab,
+          options.processorOptions.capacity,
+          this._sampleRate,
+          this._channelCount,
+          this._fSampleCount,
+        );
+      } else {
+        this._audioBufferWriter = WritableAudioBuffer.allocate(
+          options.processorOptions.bufferSec,
+          this._sampleRate,
+          this._channelCount,
+          this._fSampleCount,
+        );
+        this._audioBuffer = new ReadableAudioBuffer(
+          this._audioBufferWriter.buffer,
+          this._audioBufferWriter.bufferCapacity,
```

Comment:
ReadableAudioBuffer is created always in both **if** and **else**, but with different parameters. May be it has sense to move it out of if-else and just use condition to choose parameters?